src/lib/leaflet-utils/scenario-search-results.js

```diff
@@ -0,0 +1,36 @@
+import { getFeatureIdByScenarioId } from "../load-breach";
+
+export async function getScenarioSearchResults(query, { selectFeatureMode, mapId }) {
+  const queries = query.split(',')
+  if (queries && queries.every(q => isNaN(q)) || selectFeatureMode === 'disabled') return []
+  return Promise.all(
+    queries.map(id => {
+      const trimmedId = id.trim()
+      if (isNaN(trimmedId)) return;
+      return getFeatureIdByScenarioId(Number(trimmedId)).then((result) => ({
+        name: result.breachlocationname,
+        id: result.scenarioId,
+        link: `/${selectFeatureMode === 'single' ? 'scenarios' : 'combine'}/${mapId}/${result.scenarioId}`,
+        html: `<span>${result.breachlocationname}</span><br/><span class="leaflet-control-geocoder-address-context">Scenario id: ${result.scenarioId}</span>`,
```

Comment:
Maybe overkill in this situation, but let's escape the injected parts of the HTML

Reply:
What do you mean exactly?